scripts/workers/worker_utils.py

```diff
@@ -1,92 +1,187 @@
+import contextlib
 import logging
+import os
 import requests
+import signal
+import sys
+import traceback
+from os.path import join
+import zipfile
+
+AUTH_TOKEN = os.environ.get("AUTH_TOKEN")
 
 logger = logging.getLogger(__name__)
 
 
-URLS = {
-    "get_message_from_sqs_queue": "/api/jobs/challenge/queues/{}/",
-    "delete_message_from_sqs_queue": "/api/jobs/queues/{}/",
-    "get_submission_by_pk": "/api/jobs/submission/{}",
-    "get_challenge_phases_by_challenge_pk": "/api/challenges/{}/phases/",
-    "get_challenge_by_queue_name": "/api/challenges/challenge/queues/{}/",
-    "get_challenge_phase_by_pk": "/api/challenges/challenge/{}/challenge_phase/{}",
-    "update_submission_data": "/api/jobs/challenge/{}/update_submission/",
-}
+@contextlib.contextmanager
+def stdout_redirect(where):
+    sys.stdout = where
+    try:
+        yield where
+    finally:
+        sys.stdout = sys.__stdout__
+
+
+@contextlib.contextmanager
+def stderr_redirect(where):
+    sys.stderr = where
+    try:
+        yield where
+    finally:
+        sys.stderr = sys.__stderr__
+
+
+def alarm_handler(signum, frame):
+    raise ExecutionTimeLimitExceeded
+
+
+class GracefulKiller:
+    kill_now = False
+
+    def __init__(self):
+        signal.signal(signal.SIGINT, self.exit_gracefully)
+        signal.signal(signal.SIGTERM, self.exit_gracefully)
+
+    def exit_gracefully(self, signum, frame):
+        self.kill_now = True
+
+
+class ExecutionTimeLimitExceeded(Exception):
+    pass
 
 
 class EvalAI_Interface:
     def __init__(self, AUTH_TOKEN, EVALAI_API_SERVER, QUEUE_NAME):
         self.AUTH_TOKEN = AUTH_TOKEN
         self.EVALAI_API_SERVER = EVALAI_API_SERVER
         self.QUEUE_NAME = QUEUE_NAME
+        self.URLS = {
+            "get_message_from_sqs_queue": "/api/jobs/challenge/queues/{}/",
+            "delete_message_from_sqs_queue": "/api/jobs/queues/{}/",
+            "get_submission_by_pk": "/api/jobs/submission/{}",
+            "get_challenge_phases_by_challenge_pk": "/api/challenges/{}/phases/",
+            "get_challenge_by_queue_name": "/api/challenges/challenge/queues/{}/",
+            "get_challenge_phase_by_pk": "/api/challenges/challenge/{}/challenge_phase/{}",
+            "update_submission_data": "/api/jobs/challenge/{}/update_submission/",
+        }
 
     def get_request_headers(self):
         headers = {"Authorization": "Token {}".format(self.AUTH_TOKEN)}
         return headers
 
     def make_request(self, url, method, data=None):
         headers = self.get_request_headers()
-        try:
-            response = requests.request(
-                method=method, url=url, headers=headers, data=data
-            )
-            response.raise_for_status()
-        except requests.exceptions.RequestException:
-            logger.info(
-                "The worker is not able to establish connection with EvalAI"
-            )
-            raise
-        return response.json()
```

Comment:
@pushkalkatara 
Can you please explain this change?

Reply:
Oops! Sorry. I missed this while rearranging between `remote_submission_worker.py` and `rl_submission_worker.py`. Thanks for pointing out. I'll revert back to the one taken from `rl_submission_worker.py`

Reply:
Fixed with the recent commit.